doc/devdocs/development/new-powertoy.md

```diff
@@ -0,0 +1,287 @@
+```
+
+7. `call_custom_action` allows custom actions to be called based on signals from the settings app.
+```c++
+void call_custom_action(const wchar_t* action) override
+```
+
+8. Lifecycle events control the whether the module is enabled or not as well as the default status of the module.
+```c++
+virtual void enable() // starts the module
+virtual void disable() // terminates the module and performs any cleanup
+virtual bool is_enabled() // returns if the module is currently enabled
+virtual bool is_enabled_by_default() const override // allows the module to dictate whether it should be enabled by default in the PowerToys app.
+```
+
+9. Hotkey functions control the status of the hotkey.
+```c++
+// takes the hotkey from settings into a format the interface can understand
+void parse_hotkey(PowerToysSettings::PowerToyValues& settings) 
+
+// returns the hotkeys from settings
+virtual size_t get_hotkeys(Hotkey* hotkeys, size_t buffer_size) override 
+
+// performs logic when the hotkey event is fired
+virtual bool on_hotkey(size_t hotkeyId) override 
+```
+
+### Notes
+* Keep module logic isolated under `/modules/<YourModule>`
+* Use shared utilities from [`common`](https://github.com/microsoft/PowerToys/tree/main/src/common) instead of cross-module dependencies
+* init/set/get config use preset functions to access the settings. Check out the [`settings_objects.h`](https://github.com/microsoft/PowerToys/blob/main/src/common/SettingsAPI/settings_helpers.h) in `src\common\SettingsAPI`
+
+---
+## 3. Bootstrapping Your Module
+1. Use the [template](https://github.com/microsoft/PowerToys/tree/main/tools/project_template) to generate the module interface starter code.
+2. Update all projects and namespaces with your module name.
+3. Update GUIDs in `.vcxproj` and solution files.
+4. Update the functions mentioned in the above section with your custom logic.
+5. In order for your module to be detected by the runner you are required to add  references to various lists. In order to register your module, add the corresponding module reference to the lists that can be found in the following files. (Hint: search other modules names to find the lists quicker)
+    * `src/runner/modules.h`
+    * `src/runner/modules.cpp`
+    * `src/runner/resource.h`
+    * `src/runner/settings_window.h`
+    * `src/runner/settings_window.cpp`
+    * `src/runner/main.cpp`
+    * `src/common/logger.h` (for logging)
+6. ModuleInterface should build your `ModuleInterface.dll`. This will allow the runner to interact with your service.
+
+> **Gotcha:** Mismatched module IDs are one of the most common causes of load failures. Keep your ID consistent across manifest, registry, and service.
+
+---
+## 4. Write your service
+This is going to look different for every PowerToy. It may be easier to develop the application independently, and then link in the PowerToys settings logic later. But you have to write the service first before connecting it to the runner.
```

Comment:
```suggestion
This is going to look different for every PowerToy. It may be easier to develop the application independently, and then link in the PowerToys settings logic later. But you have to write the service first, before connecting it to the runner.
```